src/sage/rings/finite_rings/integer_mod.pyx

```diff
@@ -1378,38 +1378,39 @@ cdef class IntegerMod_abstract(FiniteRingElement):
         EXAMPLES::
 
 
-            sage: K = GF(31)
-            sage: a = K(22)
-            sage: K(22).nth_root(7)
+            sage: K = GF(31)                                                            # optional - sage.rings.finite_rings
+            sage: a = K(22)                                                             # optional - sage.rings.finite_rings
+            sage: K(22).nth_root(7)                                                     # optional - sage.rings.finite_rings
             13
-            sage: K(25).nth_root(5)
+            sage: K(25).nth_root(5)                                                     # optional - sage.rings.finite_rings
             5
-            sage: K(23).nth_root(3)
+            sage: K(23).nth_root(3)                                                     # optional - sage.rings.finite_rings
             29
-            sage: mod(225,2^5*3^2).nth_root(4, all=True)
-            [225, 129, 33, 63, 255, 159, 9, 201, 105, 279, 183, 87, 81, 273, 177, 207, 111, 15, 153, 57, 249, 135, 39, 231]
-            sage: mod(275,2^5*7^4).nth_root(7, all=True)
+            sage: mod(225, 2^5*3^2).nth_root(4, all=True)                               # optional - sage.rings.padics
+            [225, 129, 33, 63, 255, 159, 9, 201, 105, 279, 183, 87, 81,
+             273, 177, 207, 111, 15, 153, 57, 249, 135, 39, 231]
+            sage: mod(275, 2^5*7^4).nth_root(7, all=True)                               # optional - sage.rings.padics
             [58235, 25307, 69211, 36283, 3355, 47259, 14331]
-            sage: mod(1,8).nth_root(2,all=True)
+            sage: mod(1,8).nth_root(2, all=True)                                        # optional - sage.rings.padics
             [1, 7, 5, 3]
-            sage: mod(4,8).nth_root(2,all=True)
+            sage: mod(4,8).nth_root(2, all=True)                                        # optional - sage.rings.padics
             [2, 6]
-            sage: mod(1,16).nth_root(4,all=True)
+            sage: mod(1,16).nth_root(4, all=True)                                       # optional - sage.rings.padics
             [1, 15, 13, 3, 9, 7, 5, 11]
-            sage: (mod(22,31)^200).nth_root(200)
+            sage: (mod(22,31)^200).nth_root(200)                                        # optional - sage.groups
             5
```

Comment:
Is this correct?

Reply:
Yes, it uses `sage.groups.generic`; but maybe I should ship this module with sagemath-categories